src/Tomcat/TomcatController.ts

```diff
@@ -356,11 +356,21 @@ export class TomcatController {
 
     /* tslint:disable:no-any */
     private async determineAppName(webappPath: string, server: TomcatServer): Promise<string> {
-        const defaultName: string = path.basename(webappPath, path.extname(webappPath));
-        let appName: string = defaultName;
+        const isWarFile: boolean = this.isWarFile(webappPath);
+
+        let appName: string = isWarFile
+            ? path.basename(webappPath, path.extname(webappPath))
+            : path.basename(webappPath);
+
+        // remove version information from app name if any
+        const versionIndex: number = appName.search(/\-[0-9]+\.[0-9]+/gi);
```

Comment:
the regex is not compatible with most versions like snapshot,release,beta,alpha, I suggest to also try to find the artifact at 
META-INF\maven\...\pom.xml inside the war file

Reply:
I'm not sure what you mean. The regex finds the first version part `1.0` of `1.0-SNAPSHOT` or `1.1.RELEASE` etc. removes it and what comes after it.

Reply:
I think I known you meaning, you find the index of '1.0 ' and then substr to that index, then I think you'd better modify the regex to match the last occurrence of slash and number